### PR TITLE
Track list design changes

### PR DESCRIPTION
**Pull request by kkuepper**

### CODE REVIEW

File: BMM.UI.Android/Resources/drawable/icon_download_done.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (olivereisenhut) [inline BMM.UI.Android/Resources/drawable/icon_download_done.xml:1]: Can you delete the .png versions of all the assets now added as SVG drawable?

**Review by olivereisenhut — CHANGES_REQUESTED** [review]

File: BMM.Core/BMM.Core.csproj [hunk 2]
```diff
@@ -41,5 +41,8 @@
   <ItemGroup>
     <Compile Remove="ViewModels\PodcastSection.cs" />
   </ItemGroup>
+  <ItemGroup>
+    <Folder Include="Implementations\MyTracks" />
```
[the remark this answers is not in the document]
Comment (kkuepper) [reply BMM.Core/BMM.Core.csproj:45]: removed

File: BMM.UI.Android/Application/Bindings/MvxCardCircle.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+        protected override void SetValueImpl(object target, object value)
+        {
+            var card = (CardView)target;
+            if (value is bool test && test)
+                card.Radius = 10000; // this takes pixel whereas most other units are in dp
+            // Caution: this does not work if it changes dynamically
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (kkuepper) [reply BMM.UI.Android/Application/Bindings/MvxCardCircle.cs:22]: it's used in listitem_tracklist_header.xml
`card_view:MvxBind="Visibility Visibility(ShowImage); CardCircle UseCircularImage">`

File: BMM.UI.Android/Application/Fragments/AlbumFragment.cs [hunk 4]
```diff
@@ -50,9 +29,25 @@ public override View OnCreateView(LayoutInflater inflater, ViewGroup container,
...
+            {
+                recyclerView.Adapter = new HeaderRecyclerAdapter((IMvxAndroidBindingContext)BindingContext);
+                recyclerView.HasFixedSize = true;
+
+                var layoutManager = new LinearLayoutManager(ParentActivity);
+                recyclerView.SetLayoutManager(layoutManager);
+            }
+        }
+
         public override void OnCreateOptionsMenu(IMenu menu, MenuInflater inflater)
```
[the remark this answers is not in the document]
Comment (kkuepper) [reply BMM.UI.Android/Application/Fragments/AlbumFragment.cs:51]: good catch

File: BMM.UI.Android/Application/TemplateSelectors/TrackListTemplateSelector.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
+
+namespace BMM.UI.Droid.Application.TemplateSelectors
+{
+    [Obsolete]
```
[the remark this answers is not in the document]
Comment (kkuepper) [reply BMM.UI.Android/Application/TemplateSelectors/TrackListTemplateSelector.cs:5]: deleted PodcastTemplateSelector and TrackCollectionSelector.
This selector though should not have been marked as obsolete.

File: BMM.UI.Android/Application/TemplateSelectors/PodcastTemplateSelector.cs [hunk 6]
```diff
@@ -1,14 +1,14 @@
-﻿namespace BMM.UI.Droid.Application.TemplateSelectors
+﻿using System;
+
+namespace BMM.UI.Droid.Application.TemplateSelectors
 {
+    [Obsolete]
```
[the remark this answers is not in the document]
Comment (kkuepper) [reply BMM.UI.Android/Application/TemplateSelectors/PodcastTemplateSelector.cs:5]: deleted

File: BMM.UI.Android/Application/ValueConverters/DownloadStatusVisibilityValueConverter.cs [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+            if (!(item is Track track))
+                return false;
+
+            if (IsIt(track))
+                return true;
+            else
+                return false;
+        }
+
+        protected abstract bool IsIt(Track track);
```
[the remark this answers is not in the document]
Comment (kkuepper) [reply BMM.UI.Android/Application/ValueConverters/DownloadStatusVisibilityValueConverter.cs:27]: is HasItRightStatus better understandable?

File: BMM.UI.Android/Application/ValueConverters/VisibleOrHiddenValueConverter.cs [hunk 8]
```diff
@@ -12,4 +12,13 @@ protected override MvxVisibility Convert(bool value, object parameter, CultureIn
             return value ? MvxVisibility.Visible : MvxVisibility.Hidden;
         }
     }
+
+    public class VisibilityAndValueConverter : MvxBaseVisibilityValueConverter<bool>
```
[the remark this answers is not in the document]
Comment (kkuepper) [reply BMM.UI.Android/Application/ValueConverters/VisibleOrHiddenValueConverter.cs:16]: removed

File: BMM.UI.Android/BMM.UI.Droid.csproj [hunk 9]
```diff
@@ -1,8 +1,9 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
-    <LangVersion>8</LangVersion>
+    <LangVersion>8.0</LangVersion>
     <AndroidSupportedAbis>armeabi-v7a;x86;arm64-v8a;x86_64</AndroidSupportedAbis>
+    <TargetFrameworkVersion>v5.0</TargetFrameworkVersion>
```
[the remark this answers is not in the document]
Comment (kkuepper) [reply BMM.UI.Android/BMM.UI.Droid.csproj:6]: I think I did it manually. Reverted it

File: BMM.UI.Android/Application/Bindings/MvxCardCircle.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (kkuepper) [reply BMM.UI.Android/Application/Bindings/MvxCardCircle.cs:22]: add comment

File: BMM.Api/Implementation/Models/Album.cs [hunk 10]
```diff
@@ -31,5 +31,7 @@ public Album()
         public IEnumerable<string> Tags { get; set; }
 
         public string Title { get; set; }
+
+        public string Description { get; set; }
```
Comment (kkuepper) [inline BMM.Api/Implementation/Models/Album.cs:35]: still needs to be implemented for Android

File: BMM.Api/Implementation/Models/Playlist.cs [hunk 11]
```diff
@@ -15,5 +15,6 @@ public Playlist()
 
         public string Title { get; set; }
 
+        public string Description { get; set; }
```
[the remark this answers is not in the document]
Comment (olivereisenhut) [reply BMM.Api/Implementation/Models/Playlist.cs:18]: No not yet

File: BMM.UI.iOS/Application/Helpers/GradientView.cs [hunk 12]
```diff
@@ -1,16 +1,41 @@
 using CoreAnimation;
+using CoreGraphics;
 using Foundation;
 using ObjCRuntime;
 using UIKit;
 
 namespace BMM.UI.iOS.Helpers
 {
-    //https://stackoverflow.com/questions/17555986/cagradientlayer-not-resizing-nicely-tearing-on-rotation
+    // https://stackoverflow.com/questions/17555986/cagradientlayer-not-resizing-nicely-tearing-on-rotation
```
[the remark this answers is not in the document]
Comment (olivereisenhut) [reply BMM.UI.iOS/Application/Helpers/GradientView.cs:9]: Yes

File: BMM.UI.iOS/Application/ViewController/ContributorViewController.cs [hunk 13]
```diff
@@ -22,30 +24,34 @@ public override void ViewDidLoad()
...
             set.Bind(source).For(s => s.IsFullyLoaded).To(s => s.IsFullyLoaded);
-            set.Bind(CoverImage).For(v => v.ImagePath).To(vm => vm.Contributor.Cover);
             set.Bind(CircleCoverImage).For(v => v.ImagePath).To(vm => vm.Contributor.Cover);
-            set.Bind(TrackTableTitle).To(vm => vm.TextSource).WithConversion<MvxLanguageConverter>("HeaderText");
+            set.Bind(TrackCountLabel).To(vm => vm.TrackCountString);
+            set.Bind(NameLabel).To(vm => vm.Contributor.Name);
+            set.Bind(PlayButton).To(vm => vm.ShufflePlayCommand);
+            set.Bind(PlayButton).For(v => v.BindTitle()).To(vm => vm.DocumentsTextSource).WithConversion<MvxLanguageConverter>("Shuffle");
+
+            // set.Bind(TrackTableTitle).To(vm => vm.TextSource).WithConversion<MvxLanguageConverter>("HeaderText");
```
[the remark this answers is not in the document]
Comment (olivereisenhut) [reply BMM.UI.iOS/Application/ViewController/ContributorViewController.cs:43]: Removed it

File: BMM.UI.iOS/BMM.UI.iOS.csproj [hunk 14]
```diff
@@ -405,6 +405,9 @@
     <BundleResource Include="Resources\btv_teaser.png" />
     <BundleResource Include="Resources\btv_teaser@2x.png" />
     <BundleResource Include="Resources\btv_teaser@3x.png" />
+    <BundleResource Include="Resources\icon_bell.png" />
```
[the remark this answers is not in the document]
Comment (olivereisenhut) [reply BMM.UI.iOS/BMM.UI.iOS.csproj]: Did the renaming

File: BMM.Core/ViewModels/ITrackListViewModel.cs [hunk 15]
```diff
@@ -0,0 +1,24 @@
+﻿namespace BMM.Core.ViewModels
+{
+    public interface ITrackListViewModel
```
[the remark this answers is not in the document]
Comment (olivereisenhut) [reply BMM.Core/ViewModels/ITrackListViewModel.cs:3]: The `ITrackListDisplayable` represents a document inside of a list which has a title and a cover. I don't think there is a benefit by putting everything into a mega Interface which destroys the benefits of interfaces which with you can enforce as little logic as needed and combine them.

[the remark this answers is not in the document]
Comment (kkuepper) [reply BMM.Core/ViewModels/ITrackListViewModel.cs:3]: When it's not the same, then we shouldn't do it

File: BMM.UI.iOS/Application/ViewController/PodcastViewController.cs [hunk 16]
```diff
@@ -37,57 +49,56 @@ public override void ViewDidLoad()
...
+
+            set.Bind(FollowButton).For(v => v.Hidden).To(vm => vm.IsFollowing);
             set.Bind(FollowButton).To(vm => vm.ToggleFollowingCommand);
-            set.Bind(FollowingTickImageView).For("Visibility").To(vm => vm.IsFollowing).WithConversion(new InvertedVisibilityConverter());
-            set.Bind(FollowingButton).For("Visibility").To(vm => vm.IsFollowing).WithConversion(new InvertedVisibilityConverter());
-            set.Bind(FollowButton).For("Visibility").To(vm => vm.IsFollowing);
-            set.Bind(ButtonLabel).To(vm => vm.FollowButtonText);
+            set.Bind(FollowButton).For(v => v.BindTitle()).To(vm => vm.TextSource).WithConversion<MvxLanguageConverter>("Follow");
+
+            // todo I think in podcasts it should be play instead of shuffle
```
[the remark this answers is not in the document]
Comment (kkuepper) [reply BMM.UI.iOS/Application/ViewController/PodcastViewController.cs:63]: replace it with Play action instead

[the remark this answers is not in the document]
Comment (olivereisenhut) [reply BMM.UI.iOS/Application/ViewController/PodcastViewController.cs]: Implementation done for iOS. Can you do Android @kkuepper?

[the remark this answers is not in the document]
Comment (kkuepper) [reply BMM.UI.iOS/Application/ViewController/PodcastViewController.cs]: Android is done

**Pull request merged**